libres/lib/include/ert/res_util/string.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <string>
+#include <string_view>
+
+namespace ert {
+
+/**
+ * Non memory allocating string splitting function
+ *
+ * \note The input string is a view. It is therefore Undefined Behaviour to
+ * modify the original string from inside the `func` parameter.
+ */
+template <typename Func>
+void split(std::string_view str, char delimiter, Func &&func) {
```

Comment:
Had to read this a few times and now that I have convinced myself that I understand it, I like!